lightning/src/routing/router.rs

```diff
@@ -1520,12 +1530,89 @@ where L::Target: Logger {
+fn add_random_cltv_offset(route: &mut Route, payment_params: &PaymentParameters, network: &NetworkGraph, random_seed_bytes: &[u8; 32]) {
+	let network_graph = network.read_only();
+	let network_channels = network_graph.channels();
+	let network_nodes = network_graph.nodes();
+
+	for path in route.paths.iter_mut() {
+		let mut shadow_ctlv_expiry_delta_offset: u32 = 0;
+
+		// Choose the last publicly known node as the starting point for the random walk
+		if let Some(starting_hop) = path.iter().rev().find(|h| network_nodes.contains_key(&NodeId::from_pubkey(&h.pubkey))) {
+			let mut cur_node_id = NodeId::from_pubkey(&starting_hop.pubkey);
+
+			// Init PRNG with path nonce
+			let mut path_nonce = [0u8; 8];
+			path_nonce.copy_from_slice(&cur_node_id.as_slice()[..8]);
+			let mut prng = ChaCha20::new(random_seed_bytes, &path_nonce);
+			let mut random_path_bytes = [0u8; ::core::mem::size_of::<usize>()];
+
+			// Pick a random path length in [1 .. 3]
+			prng.process_in_place(&mut random_path_bytes);
+			let random_walk_length = usize::from_be_bytes(random_path_bytes).wrapping_rem(3).wrapping_add(1);
+
+			for _random_hop in 0..random_walk_length { 
+				if let Some(cur_node) = network_nodes.get(&cur_node_id) {
+					// Randomly choose the next hop
+					prng.process_in_place(&mut random_path_bytes);
+					if let Some(random_index) = usize::from_be_bytes(random_path_bytes).checked_rem(cur_node.channels.len()) {
+						if let Some(random_channel_id) = cur_node.channels.get(random_index) {
+							if let Some(random_channel) = network_channels.get(random_channel_id) {
+								if random_channel.node_one == cur_node_id {
+									cur_node_id = random_channel.node_two;
+									if let Some(dir_info) = &random_channel.one_to_two {
+										shadow_ctlv_expiry_delta_offset = shadow_ctlv_expiry_delta_offset
+											.checked_add(dir_info.cltv_expiry_delta.into()).unwrap_or(shadow_ctlv_expiry_delta_offset);
+									}
+								} else {
+									cur_node_id = random_channel.node_one;
+									if let Some(dir_info) = &random_channel.two_to_one {
+										shadow_ctlv_expiry_delta_offset = shadow_ctlv_expiry_delta_offset
+											.checked_add(dir_info.cltv_expiry_delta.into()).unwrap_or(shadow_ctlv_expiry_delta_offset);
+									}
+								};
+							}
+						}
+					}
+				}
+			}
+		} else {
+			// If the entire path is private, choose a random offset from multiples of 40, i.e., the most used cltv_expiry_delta
+			let mut prng = ChaCha20::new(random_seed_bytes, &[0u8; 8]);
+			let mut random_bytes = [0u8; 4];
+			prng.process_in_place(&mut random_bytes);
+			let random_walk_length = u32::from_be_bytes(random_bytes).wrapping_rem(3).wrapping_add(1);
+			shadow_ctlv_expiry_delta_offset = random_walk_length.wrapping_mul(40);
+		}
+
+		// Limit the total offset to reduce the worst-case locked liquidity timevalue
+		const MAX_SHADOW_CLTV_EXPIRY_DELTA_OFFSET: u32 = 432; // 3*144
```

Comment:
Replace `432` with the calculation instead of using a comment.

Reply:
Addressed by https://github.com/lightningdevkit/rust-lightning/commit/386a0b553130173f576ecc89424fcd6791a41dcc.